File: TypeScript/count_odd_num/1523.ts
function countOdds(low: number, high: number): number {
  const length = high - low + 1;
  let result = Math.floor(length / 2);
  if (low % 2 == 1 && high % 2 == 1) {
    return ++result;
  }
  return result;
}

export default countOdds;

File: Go/count_odd_number/1523.go
package countoddnumber

func countOdds(low int, high int) int {
	count := 0
	for i := low; i <= high; i++ {
		if i%2 != 0 {
			count++
		}
	}
	return count
}

// Second solution
func countOdds2(low int, high int) int {
	length := high - low + 1
	result := length / 2
	if low%2 == 1 && high%2 == 1 {
		result++
	}
	return result
}
